server/src/main/java/org/opensearch/cluster/ClusterInfo.java

```diff
@@ -117,6 +121,39 @@ public ClusterInfo(StreamInput in) throws IOException {
         } else {
             this.nodeFileCacheStats = Map.of();
         }
+
+        calculateAvgFreeAndTotalBytes(mostAvailableSpaceUsage);
+    }
+
+    /**
+     * Returns a {@link DiskUsage} for the {@link RoutingNode} using the
+     * average usage of other nodes in the disk usage map.
+     * @param usages Map of nodeId to DiskUsage for all known nodes
+     */
+    private void calculateAvgFreeAndTotalBytes(final Map<String, DiskUsage> usages) {
```

Comment:
Do we need a test for this?

Reply:
It will automatically get covered in the exiting unit and integ tests as we are calling this during initialisation itself.